src/components/chown_component.py

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env python3
+# Copyright 2025 Ubuntu
+# See LICENSE file for licensing details.
+
+import logging
+from typing import Optional
+
+from charmed_kubeflow_chisme.components.component import Component
+from charmed_kubeflow_chisme.exceptions import ErrorWithStatus
+from ops import ActiveStatus, CharmBase, StatusBase, WaitingStatus
+
+logger = logging.getLogger(__name__)
+
+
+class ChownMountedStorageComponent(Component):
+    """A component that ensures mounted storage is owned by the user that runs the service.
+
+    Args:
+        charm(CharmBase): this charm
+        name(str): name of the component
+        storage_path(str): the path where the storage is mounted and is desired to change owners.
+        workload_user(str): the user that runs the service (workload).
+    """
+
+    def __init__(
+        self,
+        charm: CharmBase,
+        name: str,
+        storage_path: Optional[str] = "/data",
+        workload_user: Optional[str] = "_daemon_",
+    ):
+        super().__init__(charm, name)
+
+        self.storage_path = storage_path
+        self.workload_user = workload_user
+        self.charm = charm
+
+    def chown_storage_path(self, storage_path, workload_user) -> None:
+        container = self.charm.unit.get_container("mlmd-grpc-server")
```

Comment:
suggestion(non-blocking): To make `chown_component.py` potentially reusable with zero changes, we could introduce a `workload_container_name` field in  `charm.py` and use that instead of specifically `mlmd-grpc-server`
```suggestion
        container = self.charm.unit.get_container(self.charm.workload_container_name)
```
```suggestion
        container = self.charm.unit.get_container("mlmd-grpc-server")
```